subprojects/common/analysis/src/main/java/hu/bme/mit/theta/analysis/algorithm/mcm/analysis/PartialSolver.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ *  Copyright 2023 Budapest University of Technology and Economics
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package hu.bme.mit.theta.analysis.algorithm.mcm.analysis
+
+import hu.bme.mit.theta.common.Tuple
+import hu.bme.mit.theta.graphsolver.ThreeVL
+import hu.bme.mit.theta.graphsolver.collectSubRelations
+import hu.bme.mit.theta.graphsolver.compilers.GraphPatternCompiler
+import hu.bme.mit.theta.graphsolver.patterns.constraints.GraphConstraint
+import hu.bme.mit.theta.graphsolver.solvers.GraphSolver
+
+class PartialSolver<T>(
```

Comment:
Please add documentation (or links to papers) that describe what this algorithm can be used for.